### ISSUE TITLE
confirmSetupIntent doesn't return any callback(promise)

### ISSUE THREAD

**Issue by azmicirit** [stripe/stripe-react-native#1180]
**Describe the bug**
`confirmSetupIntent` hook doesn't return any promise in android but new card is adding successfully in stripe.
I'm getting `paymentMethod` result from `createPaymentMethodand` hook and its id is correct and call the `confirmSetupIntent` method from `useStripe` hook and also it adds a new card into customer's file in stripe but it never returns any result.

There is no error/warning in Android logs.
I haven't tested it out on iOS device.

```
  const { createPaymentMethod, confirmSetupIntent } = useStripe();

  const saveCard = async () => {
    setLoading(true);

    const createPaymentMethodResult = await createPaymentMethod({
      paymentMethodType: 'Card',
    });

    const paymentMethod = createPaymentMethodResult?.paymentMethod;

    if (!paymentMethod) {
      // ERROR - this part is working fine
    } else {
      const clientSecret = await createPaymentCard(paymentMethod.id);

      if (createSetupIntentResult.success) {
        const { error: confirmSetupIntentError, setupIntent } = await confirmSetupIntent(clientSecret, {
          paymentMethodType: 'Card',
          paymentMethodData: {
            paymentMethodId: paymentMethod.id,
          },
        });

        // UNREACHABLE
        console.log('error', confirmSetupIntentError);
      } 
    }
  };
```

**Expected behavior**
Catching error of confirmSetupIntent or setupIntent after call confirmSetupIntent method.

**Desktop (please complete the following information):**
 - OS: Macos M1 12.6
 - React: 0.64.0
 - Version 0.17.0

**Smartphone (please complete the following information):**
 - Device: Xiaomi Readmi 10 MUI 13

**Additional context**

**Comment by yarikpwnzer** [comment]
Some of our users are facing same issue, personally I can't reproduce on emulator or real device.

**Comment by charliecruzan-stripe** [comment]
Hi! this was fixed in v0.18.1
